src/main/java/core/mate/academy/service/MachineProducerStrategy.java

```diff
@@ -0,0 +1,24 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Machine;
+import core.mate.academy.service.producerimpl.BulldozerProducerImpl;
+import core.mate.academy.service.producerimpl.ExcavatorProducerImpl;
+import core.mate.academy.service.producerimpl.TruckProducerImpl;
+import java.util.ArrayList;
+
+public class MachineProducerStrategy {
+    public MachineProducer<? extends Machine> getMachineProducerStrategy(
+            Class<? extends Machine> type) {
+        switch (type.getSimpleName()) {
+            case "Bulldozer":
+                return new BulldozerProducerImpl();
+            case "Excavator":
+                return new ExcavatorProducerImpl();
+            case "Truck":
+                return new TruckProducerImpl();
+            default:
+                return (MachineProducer<Machine>) ArrayList::new;
```

Comment:
okay, so what will happen when you approach `default` branch?

`MachineProducer<Machine>` is definitely not a `Supplier<ArrayList>` so I suppose you'll get `ClassCastException`

how about returning one of the existing producers?

Reply:
This is the method refference and we have some check for wrong type - all tests passed. Ok, I changed it to full anonymous implementation ). 
"how about returning one of the existing producers?" We have some limits on this in Common mistakes.

Reply:
and really don't understand why I have to create default Machine producer implementation with method get wchich return empty list. In this case it is really better to use some anonymous impl (method refference or lambda) and return empty list. Ok, but if you really consist on it I will create some class like 
class MachineProducerimpl{
@Override
 method get() {
return empty list;
}
}

=)

Reply:
oh wait...

seems like I'm wrong here :) 

`ClassCastException` will not happen, since ArrayList::new lambda suits to MachineProducer (cause it's a functional interface)

you've implemented a nice trick,  which I didn't even notice at the first time

great job!)